src/query.rs

```diff
@@ -206,8 +247,45 @@ impl OverrideStack {
     ///
     /// The inserted overrides will take precedence over any lower item in the stack,
     /// if both maps have a not-`None` entry for a given lint.
-    pub fn push(&mut self, item: &OverrideMap) {
-        self.0.push(item.clone());
+    ///
+    /// # Errors
+    ///
+    /// Returns an `Err` if there is a configuration conflict in the passed map: when
+    /// multiple entries in `item` configure the same entry for the same lint. To handle
+    /// configuration precedence in a defined way, call this function multiple times, and
+    /// entries at the top of the stack will override entries towards the bottom of the stack
+    /// when there is a conflict.
+    pub fn try_push(&mut self, item: &OverrideMap) -> anyhow::Result<()> {
+        let mut compiled_map = BTreeMap::<_, QueryOverride>::new();
+
+        for (id, overrides) in item {
+            let lints = id.lints(ALL_QUERIES.get_or_init(SemverQuery::all_queries));
+
+            for lint in lints {
+                if let Some(entry) = compiled_map.get_mut(lint) {
+                    if let Some(lint_level) = overrides.lint_level {
+                        if let Some(old) = entry.lint_level {
+                            anyhow::bail!("Configuration conflict detected for lint `{lint}`:\n\
+                                lint level is set to `{old:?}` and {lint_level:?} at the same config priority.");
```

Comment:
This is unfortunately a supremely non-useful error message: it tells the user "go figure it out" without actually saying which entries are the ones at that priority. The entries might not be easy to find, since they might be on the lint itself, or any of its groups.

This would be very frustrating to encounter.